src/PartialNameMatchHandler.ts

```diff
@@ -0,0 +1,85 @@
+import { Context, SQSBatchResponse, SQSEvent, SQSRecord } from "aws-lambda";
+import { Logger } from "@aws-lambda-powertools/logger";
+import { Metrics } from "@aws-lambda-powertools/metrics";
+import { LambdaInterface } from "@aws-lambda-powertools/commons";
+import { Constants, EnvironmentVariables } from "./utils/Constants";
+import { failEntireBatch, passEntireBatch } from "./utils/SqsBatchResponseHelper";
+import { PartialNameProcessor } from "./services/PartialNameProcessor";
+import { PutObjectCommand, PutObjectCommandInput, S3Client } from "@aws-sdk/client-s3";
+import { NodeHttpHandler } from "@aws-sdk/node-http-handler";
+import { checkEnvironmentVariable } from "./utils/EnvironmentVariables";
+import { absoluteTimeNow } from "./utils/DateTimeUtils";
+
+const { POWERTOOLS_METRICS_NAMESPACE = Constants.BAV_METRICS_NAMESPACE, POWERTOOLS_LOG_LEVEL = "DEBUG", POWERTOOLS_SERVICE_NAME = Constants.PARTIAL_NAME_MATCH_HANDLER } = process.env;
+
+export const logger = new Logger({
+	logLevel: POWERTOOLS_LOG_LEVEL,
+	serviceName: POWERTOOLS_SERVICE_NAME,
+});
+
+const metrics = new Metrics({ namespace: POWERTOOLS_METRICS_NAMESPACE, serviceName: POWERTOOLS_SERVICE_NAME });
+
+let s3Client: S3Client;
+
+class PartialNameMatchHandler implements LambdaInterface {
+	@metrics.logMetrics({ throwOnEmptyMetrics: false, captureColdStartMetric: true })
+	async handler(event: SQSEvent, context: Context): Promise<SQSBatchResponse> {
+
+		if (!s3Client) {
+			s3Client = new S3Client({
+				region: process.env.REGION,
```

Comment:
I'm not sure if I'm being blind but I can't find the `REGION` env var?

Reply:
It gets it from AWS runtime pretty sure I can log it out to confirm